RecoDataProducts/src/IntensityInfoCalo.cc

```diff
@@ -0,0 +1,41 @@
+// Implementation of methods
+#include "Offline/RecoDataProducts/inc/IntensityInfoCalo.hh"
+#include "cetlib_except/exception.h"
+#include <algorithm>
+
+namespace mu2e {
+
+  // Store a Caphri hit
+  void IntensityInfoCalo::addCaphriHit(const double energy, const int ID) {
+
+    // Map the crystal ID to an index between 0-3
+    const auto itr = std::find(CaloConst::_caphriId.begin(), CaloConst::_caphriId.end(), ID);
+    if(itr == CaloConst::_caphriId.end()) throw cet::exception("RECO") << "Crystal ID " << ID << " is not listed as a CAPHRI crystal!"; // not a CAPHRI crystal
+    const unsigned short index = std::distance(CaloConst::_caphriId.begin(), itr);
+    if(index > 3) throw cet::exception("RECO") << "CAPHRI crystal index " << index << "(crystal ID " << ID << ") is greater than 3!"; // must be between 0-3
+
+    const unsigned short e_short = energy / caphriEnergyUnits_; // store in different units from MeV input to adjust the precision
+
+    // encode the energy + index into a single short
+    const unsigned short hit_info = e_short | index << caphriIndexBits_;
+
+    // add it to the hit vector
+    caphriHits_.push_back(hit_info);
+  }
+
+  // Decode CAPHRI hit information
+  void IntensityInfoCalo::getCaphriHit(const size_t ihit, double& energy, int& ID) const {
+    ID = -1; energy = 0.;
+    if(ihit >= caphriHits_.size()) throw cet::exception("RECO") << "Hit index " << ihit << " is >= N(hits) = " << caphriHits_.size();
+
+    const auto hit_info = caphriHits_[ihit];
+
+    constexpr static unsigned short caphriIndexMask = 0x3 << caphriIndexBits_; // mask for where the hit index is stored
```

Comment:
Is there a reason this is defined static here, while the companion bitshift constexpr is defined in the .hh file?